### ISSUE TITLE
How to use custom image?

### ISSUE THREAD

**Issue by xikunyu** [Opster/opensearch-k8s-operator#161]
I need to use snapshot on opensearch cluster. I found the method from https://opensearch.org/docs/latest/opensearch/snapshot-restore/#amazon-s3

And then I build my image and set image in my cluster.yaml. But it doesn't work.

```
[root@eks-test opensearch]# cat cluster.yaml
apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
metadata:
  name: opensearch
  namespace: opensearch-system
spec:
  general:
    serviceName: opensearch
    version: 2.0.0
    image: 516915001847.dkr.ecr.cn-northwest-1.amazonaws.com.cn/opensearch:2.0.0
  security:
    config:  # Everything related to the securityconfig
      securityConfigSecret:
        name:  securityconfig-secret
    tls:
      transport:
        generate: true
      http:
        generate: true
      #adminSecret:
      #  name:  # Name of a secret that contains the admin client certificate
      #adminCredentialsSecret:
      #  name:  admin-secret
  dashboards:
    enable: true
    version: 2.0.0
    replicas: 1
    #additionalConfig:
    #  opensearch_security.multitenancy.enabled: "true"
    resources:
      requests:
         memory: "512Mi"
         cpu: "200m"
      limits:
         memory: "512Mi"
         cpu: "200m"
  nodePools:
    - component: masters-data-hot
      replicas: 2
#      image: 516915001847.dkr.ecr.cn-northwest-1.amazonaws.com.cn/opensearch:2.0.0
      diskSize: "5Gi"
      NodeSelector:
      resources:
         requests:
            memory: "1Gi"
            cpu: "500m"
         limits:
            memory: "1Gi"
            cpu: "500m"
      roles:
        - "data"
        - "master"
        - "ingest"
      additionalConfig:
        node.attr.type: hot
    - component: masters-data-cold
      replicas: 1
#      image: 516915001847.dkr.ecr.cn-northwest-1.amazonaws.com.cn/opensearch:2.0.0
      diskSize: "5Gi"
      NodeSelector:
      resources:
         requests:
            memory: "1Gi"
            cpu: "500m"
         limits:
            memory: "1Gi"
            cpu: "500m"
      roles:
        - "data"
        - "master"
        - "ingest"
      additionalConfig:
        node.attr.type: cold

```
 
pod description:
```
Containers:
  opensearch:
    Container ID:   docker://6cae87b46bbc95303e98e6327ec110e358db620c872a3591d9906dafc9c1f016
    Image:          docker.io/opensearchproject/opensearch:2.0.0
    Image ID:       docker-pullable://opensearchproject/opensearch@sha256:4f49474dcb7345af0afe976221c1885dce2ceb7093516538b08cdd3bdfc33f20
    Ports:          9200/TCP, 9300/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Running
      Started:      Wed, 01 Jun 2022 16:13:30 +0800
    Ready:          False
```

**Comment by swoehrl-mw** [comment]
Hi @xikunyu. Thanks for this report. I'm pretty sure this is the same bug that was fixed in #152 but has not yet landed in a release.

**Comment by idanl21** [comment]
Hey @xikunyu, we added a DefaultRepo key on the general section, that will hellp you to use custom image.
let us know if you need any help
